upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1397,14 +1398,26 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 	config.Channels = n.channels
 	config.EtcdManifests = n.etcdManifests[role]
 
-	if cluster.Spec.ContainerRuntime == "containerd" {
-		config.ContainerdConfig = cluster.Spec.Containerd
+	if ig.Spec.Containerd != nil || cluster.Spec.ContainerRuntime == "containerd" {
+		config.ContainerdConfig = n.buildContainerdConfig(ig)
 	}
 
 	if cluster.Spec.Containerd != nil && cluster.Spec.Containerd.NvidiaGPU != nil {
 		config.NvidiaGPU = cluster.Spec.Containerd.NvidiaGPU
 	}
 
+	if ig.Spec.Containerd != nil && ig.Spec.Containerd.NvidiaGPU != nil {
+		if config.NvidiaGPU == nil {
+			config.NvidiaGPU = ig.Spec.Containerd.NvidiaGPU
+		} else {
+			reflectutils.JSONMergeStruct(&config.NvidiaGPU, ig.Spec.Containerd.NvidiaGPU)
+		}
+	}
+
+	if config.NvidiaGPU != nil && config.NvidiaGPU.DriverPackage == "" {
+		config.NvidiaGPU.DriverPackage = "nvidia-headless-460-server"
```

Comment:
we should perhaps store this const to same common place. This is now defined in two files as hardcoded, which would be the correct place to store this information? there are wellknownports etc should we add something new?

Reply:
It could certainly be moved to a const. Perhaps in the API package.

Here we are doing the merging in two locations as well. Didn't it work to move this to a function similar to how we merge workpool?